app/controllers/users/passwords_controller.rb

```diff
@@ -0,0 +1,62 @@
+class Users::PasswordsController < Devise::PasswordsController
+  include ApplicationHelper
+  include PhoneNumberHelper
+  include SmsBodyHelper
+
+  def create
+    email = params[resource_name][:email]
+    phone_number = params[resource_name][:phone_number]
+    reset_token = nil
+
+    if email.blank? && phone_number.blank?
```

Comment:
I think you should have 2 clear paths for email and sms based on what is filled out. Right now, its hard to follow the logic as we have a lot `if` statements that keep checking `email.blank?` and `phone_number.blank?`. Maybe:

```ruby
if !email.blank? && valid_email
   # run this code which  only has to do with the email
end

if !phone_number.blank? && valid_phone_number
   # run this code which  only has to do with the sms
end
```